### PR TITLE
Adding pyqir-evaluator example with result stream usage.

### PR DESCRIPTION
**Pull request by idavis**

### CODE REVIEW

File: eng/psakefile.ps1
Comment (bettinaheim) [inline eng/psakefile.ps1:403]: What was the reason to only look at the first case, opposed to comparing all 4 cases?

Comment (idavis) [reply eng/psakefile.ps1:403]: They are smoke tests to make sure the samples run without errors and generate some expected output.

File: examples/evaluator/README.md
Comment (bettinaheim) [inline examples/evaluator/README.md:36]: It may be worth calling out that this functionality is intended for testing purposes. It may otherwise confuse some people that are trying to use this for simulation or connecting to hardware.

File: eng/psakefile.ps1
Comment (bettinaheim) [reply eng/psakefile.ps1:403]: Since the code that we point people to runs for all 4 cases, I think it would make sense to also just include the 4 cases here. It's only a tiny tweak in the build script. I see that the PR is still in draft, so I'll hold off with approving - let me know when it is ready to be reviewed.

Comment (idavis) [reply eng/psakefile.ps1:436]: It isn't a simple change. The last bits of the output change based on the result stream. We have an exhaustive teleport tests which verify all of this behavior. The `run-examples` task verifies that the samples run and don't generate errors. If we want to have all of the examples fully verified during the build, we should put that on the backlog as a new task.

Comment (bettinaheim) [reply eng/psakefile.ps1:434]: Alright. I realized I never updated my comment after clarifying offline with you whether we have test coverage. As said there, I'll leave the judgement call up to you, as long as we have unit tests for it somewhere.

**Review by bettinaheim — APPROVED** [review]

**Pull request merged**
